src/module.c

```diff
@@ -13918,6 +13918,61 @@ RedisModuleString *RM_DefragRedisModuleString(RedisModuleDefragCtx *ctx, RedisMo
     return activeDefragStringOb(str);
 }
 
+/* Defrag callback for radix tree iterator, called for each node,
+ * used in order to defrag the nodes allocations. */
+int moduleDefragRaxNode(raxNode **noderef) {
+    raxNode *newnode = activeDefragAlloc(*noderef);
+    if (newnode) {
+        *noderef = newnode;
+        return 1;
+    }
+    return 0;
+}
+
+/* Defrag a RedisModuleDict previously allocated by RM_CreateDict. */
+typedef void *(*RedisModuleDefragDictValueCallback)(void *data, unsigned char *key, size_t keylen);
```

Comment:
@oranagra does key and keylen because data might be referenced externally?

Reply:
i think because that's what RAX keeps, and it's most likely that the module doesn't need the key in this callback at all, so it's a pity to copy it to a string object.